network/proc/findpid.go

```diff
@@ -5,99 +5,100 @@ package proc
 import (
 	"fmt"
 	"os"
-	"sort"
-	"strconv"
 	"sync"
-	"syscall"
+	"time"
 
 	"github.com/safing/portmaster/network/socket"
 
 	"github.com/safing/portbase/log"
 )
 
 var (
-	// pidsByUserLock is also used for locking the socketInfo.PID on all socket.*Info structs.
-	pidsByUserLock sync.Mutex
-	pidsByUser     = make(map[int][]int)
+	socketInfoLock sync.RWMutex
+
+	baseWaitTime  = 3 * time.Millisecond
+	lookupRetries = 3
 )
 
 // GetPID returns the already existing pid of the given socket info or searches for it.
 // This also acts as a getter for socket.*Info.PID, as locking for that occurs here.
 func GetPID(socketInfo socket.Info) (pid int) {
-	pidsByUserLock.Lock()
-	defer pidsByUserLock.Unlock()
-
-	if socketInfo.GetPID() != socket.UnidentifiedProcessID {
-		return socketInfo.GetPID()
+	// Get currently assigned PID to the socket info.
+	socketInfoLock.RLock()
+	currentPid := socketInfo.GetPID()
+	socketInfoLock.RUnlock()
```

Comment:
Why do you use a global mutex for accessing `GetPID()` from a struct thats defined in a completely different package? sockeInfo needs to holds the struct itself otherwise we will always have race-conditions because nobody outside of the `proc` package can use this lock